### PR TITLE
Handshake and LocalHandshake Tracers

### PR DESCRIPTION
**Pull request by karknu**
Issue
-----------
#623 

Checklist
---------
- [ x] This PR contains all the work required to resolve the linked issue.

- [x] The work contained has sufficient documentation to describe what it does and how to do it.

- [x] The work has sufficient tests and/or testing.

- [x] I have committed clear and descriptive commits. Be considerate as somebody else will have to read these.

- [ ] I have added the appropriate labels to this PR.

### CODE REVIEW

File: configuration/configuration-mainnet.yaml [hunk 1]
```diff
@@ -177,3 +177,9 @@ TraceTxOutbound: False
 
 # Trace TxSubmission protocol messages.
 TraceTxSubmissionProtocol: False
+
```
Comment (Jimbo4350) [inline configuration/configuration-mainnet.yaml]: Please propagate `TraceHandshake` & `TraceLocalHandshake` to all the configuration yaml files

File: cardano-config/src/Cardano/Config/Types.hs [hunk 2]
```diff
@@ -199,6 +199,8 @@ instance FromJSON NodeConfiguration where
                               <*> v .:? "TraceTxInbound" .!= False
                               <*> v .:? "TraceTxOutbound" .!= False
                               <*> v .:? "TraceTxSubmissionProtocol" .!= False
+                              <*> v .:? "TraceHandshake" .!= False
```
Comment (Jimbo4350) [inline cardano-config/src/Cardano/Config/Types.hs]: Please put `TraceHandshake` & `TraceLocalHandshake`  in alphabetical order like the rest of the tracers.

File: configuration/log-configuration.yaml [hunk 3]
```diff
@@ -181,3 +181,9 @@ TraceTxOutbound: True
 
 # Trace TxSubmission protocol messages.
 TraceTxSubmissionProtocol: True
+
+# Trace Handshake protocol messages.
```
Comment (Jimbo4350) [inline configuration/log-configuration.yaml]: Sorry @karknu these also need to be in alphabetical order.

Comment (karknu) [reply configuration/log-configuration.yaml]: The time sorting trace options would have been better spent  coming up with a better solution to having 15 near identical yaml files.

Comment (Jimbo4350) [reply configuration/log-configuration.yaml]: Agreed, these yaml files will reduce once the benchmarking stuff is out of the repo.

**Comment by karknu** [comment]
bors +r

**Pull request merged**